allennlp/predictors/open_information_extraction.py

```diff
@@ -60,6 +60,72 @@ def make_oie_string(tokens: List[Token], tags: List[str]) -> str:
 
     return " ".join(frame)
 
+def get_predicate_indices(tags: List[str]) -> List[int]:
+    """
+    Return the word indices of a predicate in BIO tags.
+    """
+    return [ind for (ind, tag)
+            in enumerate(tags) if 'V' in tag]
+
+def get_predicate_text(sent_tokens: List[Token], tags: List[str]) -> str:
+    """
+    Get the predicate in this prediction.
+    """
+    return " ".join([sent_tokens[pred_id].text
+                     for pred_id in get_predicate_indices(tags)])
+
+def check_predicates_subsumed(tags1: List[str], tags2: List[str]) -> bool:
+    """
+    Tests whether the predicate in BIO tags1 are subsumed in
+    those of tags2.
+    """
+    # Get predicate word indices from both predictions
+    pred_ind1, pred_ind2 = map(get_predicate_indices,
+                               [tags1, tags2])
+
+    # Return if pred_ind1 is contained in pred_ind2
+    return set(pred_ind1) < set(pred_ind2)
+
+def merge_predictions(tags1: List[str], tags2: List[str]) -> List[str]:
+    """
+    Merge two predictions into one. Assumes the predicate in tags1 are contained in
+    the predicate of tags2.
+    """
+    # Allow tags1 to add elements to tags2
+    return [tag2 if (tag2 != 'O')\
+            else tag1
+            for (tag1, tag2) in zip(tags1, tags2)]
+
+def consolidate_predictions(outputs: List[List[str]], sent_tokens: List[Token]) -> Dict[str, List[str]]:
+    """
```

Comment:
is it possible to have (say) predicate_a < predicate_b < predicate_c ?
it seems like your logic wouldn't work in that case